mmv1/third_party/terraform/services/alloydb/resource_alloydb_instance_test.go

```diff
@@ -406,3 +406,66 @@ resource "google_service_networking_connection" "vpc_connection" {
 }
 `, context)
 }
+
+// This test passes if we are able to create a primary instance with the network_config field defined.
+func TestAccAlloydbInstance_createInstanceWithNetworkConfigAndAllocatedIPRange(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+		"network_name":  acctest.BootstrapSharedTestNetwork(t, "alloydbinstance-networkconfig"),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckAlloydbInstanceDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAlloydbInstance_createInstanceWithNetworkConfigAndAllocatedIPRange(context),
```

Comment:
Can we add another step where we make changes to the allocate IP range and the network inside networkConfig? This will help us catch any bug for the updation use-case. Please find an example [here](https://github.com/GoogleCloudPlatform/magic-modules/blob/42aa710272092e2b1588926d081a9e1dcdcfd1c6/mmv1/third_party/terraform/services/alloydb/resource_alloydb_cluster_test.go#L10).